src/api/route-services/pipeline-response.js

```diff
@@ -0,0 +1,41 @@
+const AWS = require('../../utils/requireAWS');
+const validateRequest = require('../../utils/schema-validator');
+
+const pipelineResponse = async (io, message) => {
+  await validateRequest(message, 'PipelineResponse.v1.yaml');
+
+  // Fail hard if there was an error.
+  const { response: { error } } = message;
+
+  if (error) {
+    throw new Error(error);
+  }
```

Comment:
will the UI receive this? Or will the UI just never get response for failed pipeline steps?

Reply:
Fixed so it does receive the error.